core/chains/evm/client/client.go

```diff
@@ -29,6 +29,9 @@ type Client interface {
 	Dial(ctx context.Context) error
 	Close()
 	ChainID() *big.Int
+	// NodeStates returns a map of node ID->node state
+	// It might be nil or empty, e.g. for mock clients etc
+	NodeStates() map[int32]string
```

Comment:
Would it make sense to use the `NodeState` type instead of `int32`?

Reply:
Yeah, I was trying to avoid exposing them outside of the package but perhaps that's inevitable.